### DISCUSSION TITLE
FileOpenPicker/FileSavePicker .SuggestedStartLocation issue

### DISCUSSION THREAD

**Discussion by axtens** [microsoft/microsoft-ui-xaml#9623] [Q&A]
`FileOpenPicker` and `FileSavePicker` have a `SuggestedStartLocation` with a limited set of locations controlled by an enum. The [documentation](https://learn.microsoft.com/en-us/uwp/api/windows.storage.pickers.fileopenpicker.suggestedstartlocation?view=winrt-22621#windows-storage-pickers-fileopenpicker-suggestedstartlocation) says "The SuggestedStartLocation is not always used as the start location for the file picker. To give the user a sense of consistency, the file picker remembers the last location that the user navigated to and will generally start at that location."

Does anyone know of a third-party replacement for the Picker that has a RequiredStartLocation? We're getting really cheesed off with what happens when Windows "generally start[s] at that location".

**Comment by castorix** [comment]
> Does anyone know of a third-party replacement for the Picker that has a RequiredStartLocation? We're getting really cheesed off with what happens when Windows "generally start[s] at that location".

You can use _IFileDialog_ with  [Controlling the Default Folder](https://learn.microsoft.com/en-us/windows/win32/shell/common-file-dialog#controlling-the-default-folder) (I had added C# declarations in _CFileDialog.cs_ from [WinUI3_IFileDialog](https://github.com/castorix/WinUI3_IFileDialog) and I called _SetFolder_ in the constructor (if not null))
